File: Backend/OnBoardingDigital.Contracts/Subscription/SubscriptionResponse.cs

namespace OnBoardingDigital.Contracts.Subscription;

public record SubscriptionResponse(string Id, string FormId, string Email, DateTime CreatedAtUtc, List<SubscriptionAnswerResponse> Answers);
public record SubscriptionAnswerResponse(string Id, string FieldId, int Index, string? TextValue, bool? ChoiceValue, string? OptionsValue, long? NumberValue, string? FileName);

File: OnBoardingDigital/src/app/Dtos/subscriptionResponse.ts
export class SubscriptionResponse{
  id!: string;
  email!: string;
  formId!: string;
  createdAtUtc!: Date;
  answers!: SubscriptionAnswerResponse[];
}

export class SubscriptionAnswerResponse{
  id!: string;
  fieldId!: string;
  index!: number;
  textValue!: string | null;
  choiceValue!: boolean | null;
  optionsValue!: string | null;
  numberValue!: number | null;
  fileName!: string | null;
}


